src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -200,6 +200,43 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 		const endsWithSpace = cursorPrefix.endsWith(' ');
 		const lastWord = endsWithSpace ? '' : cursorPrefix.split(' ').at(-1) ?? '';
 
+		if (foldersRequested) {
+			if (!lastWord.trim()) {
+				resourceCompletions.push({
+					label: '.',
+					provider: 'builtin',
+					kind: TerminalCompletionItemKind.Folder,
+					isDirectory: true,
+					isFile: false,
+					detail: 'Source folder',
+					replacementIndex: cursorPosition - lastWord.length,
+					replacementLength: lastWord.length
+				});
+			}
+			if (lastWord.endsWith('..' + resourceRequestConfig.pathSeparator)) {
+				resourceCompletions.push({
+					label: lastWord + '..' + resourceRequestConfig.pathSeparator,
+					provider: 'builtin',
+					kind: TerminalCompletionItemKind.Folder,
+					isDirectory: true,
+					isFile: false,
+					replacementIndex: cursorPosition - lastWord.length,
+					replacementLength: lastWord.length
+				});
+			} else {
+				resourceCompletions.push({
+					label: '..' + resourceRequestConfig.pathSeparator,
+					provider: 'builtin',
+					kind: TerminalCompletionItemKind.Folder,
+					detail: 'Parent folder',
```

Comment:
Let's add the actual folders here since we only get this far when `cwd` is known